### PR TITLE
Add tests for Condition Service

### PR DESCRIPTION
**Pull request by schreiaj**
Stacked changes with #8759 

# BACKEND PULL REQUEST

## Related Issue

Adds tests for Condition Service

## Changes Proposed

Adds tests for Condition Service

## Additional Information

Add at least some testing for the condition service (notably to test the edge case of getting a malformed `next` link that resulted in an infinite loop

## Testing

Run the tests

### CODE REVIEW

**Review by arinkulshi-skylight — DISMISSED** [review]
Awesome tests. LGTM. You can probably merge this one instead of the other one.

**Review by kevinfiol — DISMISSED** [review]
Thanks for this, it was pretty clear to follow. :+1:

File: backend/src/main/java/gov/cdc/usds/simplereport/service/ConditionService.java [hunk 1]
```diff
@@ -37,7 +38,13 @@ public class ConditionService {
   private final ConditionRepository conditionRepository;
   private final LoincStagingRepository loincStagingRepository;
 
+  //  This is a safety check to ensure we don't get stuck in an infinite loop
+  private static final int MAX_PAGES = 1000;
+
   private static final int PAGE_SIZE = 20;
+  //  Instead of hard-coding the code names, we can use a set of code names to extract the focus
+  //  context.
+  private final HashSet<String> codeNames = new HashSet<>(List.of("focus"));
```
Comment (mpbrown) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/ConditionService.java:47]: Are we expecting this to ever have more than one value in the set? It might be more clear to specify the string "focus" in a single variable like `CONTEXT_TYPE_FOCUS = "focus"` since that refers to the context type of the ValueSet API endpoint we use to retrieve the conditions `https://tes.tools.aimsplatform.org/api/fhir/ValueSet?context-type=focus`

File: backend/src/test/java/gov/cdc/usds/simplereport/service/ConditionServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+  void setup() {
+    conditionService = new ConditionService(tesClient, conditionRepository, loincStagingRepository);
+    fhirContext = FhirContext.forR4();
+    fhirContext.getParserOptions().setStripVersionsFromReferences(false);
+    fhirContext.getParserOptions().setOverrideResourceIdWithBundleEntryFullUrl(false);
+
+    SecurityContextHolder.getContext()
+        .setAuthentication(
+            new TestingAuthenticationToken(
+                "admin", null, List.of(new SimpleGrantedAuthority("ROLE_SITE_ADMIN"))));
```
Comment (mpbrown) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/ConditionServiceTest.java:68]: We could use the `@WithSimpleReportSiteAdminUser` annotation in `SliceTestConfiguration` instead of having to manually set the authentication here. `DeviceTypeServiceTest.java` has an example of this being used, plus we could add tests to make sure standard users cannot run these syncs

File: backend/src/test/java/gov/cdc/usds/simplereport/service/ConditionServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+import org.springframework.security.authentication.TestingAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@EnableAsync
+@ActiveProfiles("test")
+class ConditionServiceTest {
```
[the remark this answers is not in the document]
Comment (DanielSass) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/ConditionServiceTest.java:49]: I'm not a huge fan of the `BaseServiceTest` paradigm and would avoid using it unless its necessary.

**Review by DanielSass — DISMISSED** [review]
